### PR TITLE
frontend: propagate errors returned by queriers

### PR DESCRIPTION
**Pull request by gouthamve**
Signed-off-by: Goutham Veeramachaneni <gouthamve@gmail.com>

### CODE REVIEW

File: pkg/querier/frontend/frontend.go [hunk 1]
```diff
@@ -201,6 +201,7 @@ func (f *Frontend) RoundTrip(r *http.Request) (*http.Response, error) {
 		// Only retry is we get a HTTP 500 or non-HTTP error.
 		if resp == nil || resp.Code/100 == 5 {
 			level.Error(f.log).Log("msg", "error processing request", "try", tries, "err", lastErr, "resp", resp)
+			lastErr = httpgrpc.ErrorFromHTTPResponse(resp)
```
Comment (tomwilkie) [inline pkg/querier/frontend/frontend.go]: resp might be `nil` here, so you probably want to break up this if.

File: pkg/querier/frontend/frontend.go [hunk 2]
```diff
@@ -217,7 +218,7 @@ func (f *Frontend) RoundTrip(r *http.Request) (*http.Response, error) {
 		return httpResp, nil
 	}
 
-	return nil, httpgrpc.Errorf(http.StatusInternalServerError, "Query failed after %d retries", f.cfg.MaxRetries)
+	return nil, httpgrpc.Errorf(http.StatusInternalServerError, "Query failed after %d retries, err: %s", f.cfg.MaxRetries, lastErr)
```
Comment (tomwilkie) [inline pkg/querier/frontend/frontend.go]: Lets just return `lastErr` if its not nil, else the old error message.

**Comment by tomwilkie** [comment]
LGTM from me, @gouthamve ?

**Comment by gouthamve** [comment]
Yes, LGTM from me :)

**Pull request merged**